2371/2/author_entity.rb

```diff
@@ -0,0 +1,43 @@
+require File.expand_path(File.dirname(__FILE__) + '/constants')
+require File.expand_path(File.dirname(__FILE__) + '/battle_entity')
+
+# AuthorEntity responsible for author info
+class AuthorEntity
+  attr_reader :author_name, :battles_info
+  def initialize(author_name, battle_name)
+    @author_name = author_name
+    @battles = []
+    @battles_info = { curse_words: 0, words_percent: 0 }
+    add_battle(battle_name)
+  end
+
+  def add_battle(battle_file_name)
+    return unless File.file?("./texts/#{battle_file_name}")
+    @battles << BattleEntity.new(battle_file_name)
+    prepare_data
+  end
+
+  def bad_words_print
```

Comment:
Treat your class like YOU are the user of the class. Thinks about **messages** which you pass to the object.
`Hey author, print me bad words!`
`Ok! author.print_bad_words`

But anyway, I don't like the idea that Author knows something about **Formatting**.
Ideally, your Author shouldn't know anything about formatting.

```ruby
class Author
  def bad_words; end # returns an array of bad words
  def bad_words_per(battle); end # returns Hash like { battle_1 => bad_words_count, battle_2 => ... }
end
```